crates/re_space_view_spatial/src/space_view_2d.rs

```diff
@@ -232,7 +232,29 @@ impl SpaceViewClass for SpatialSpaceView2D {
                     .collect(),
             }
         })
-        .unwrap_or_default()
+        .unwrap_or_default();
+
+        // TODO(jleibs): This is expensive. Would be great to track this as we build up the covering instead.
+        let remaining_entities = indicated_entities
+            .iter()
+            .filter(|entity| {
+                heuristics
+                    .recommended_space_views
+                    .iter()
+                    .all(|r| !r.query_filter.is_included(entity))
+            })
+            .collect::<Vec<_>>();
+
+        for entity in remaining_entities {
+            heuristics
+                .recommended_space_views
+                .push(RecommendedSpaceView {
+                    root: entity.clone(),
+                    query_filter: EntityPathFilter::single_entity_filter(entity),
+                });
+        }
+
+        heuristics
```

Comment:
My brain is tired. Is there a comment you can add the summarizes what is happening here, and why?